pkg/resource/fake/mocks.go

```diff
@@ -249,7 +249,11 @@ func (m *EnvironmentConfigReferencer) GetEnvironmentConfigReferences() []corev1.
 
 // ConnectionDetailsLastPublishedTimer is a mock that implements the
 // ConnectionDetailsLastPublishedTimer interface.
-type ConnectionDetailsLastPublishedTimer struct{ Time *metav1.Time }
+type ConnectionDetailsLastPublishedTimer struct {
+	// runtime.DefaultUnstructuredConverter.ToUnstructured cannot currently
+	// handle if `Time` is nil here.
+	Time *metav1.Time `json:"time,omitempty"`
```

Comment:
How about other structs whose fields don't have json tag at all? How do they work?

Reply:
Our types, when left `nil` and without the `omitempty` json tag, do not cause panics in the structured-merge-diff library because they do not have a custom string converter (which is a shortcut to roundtripping via JSON, i.e., marshal -> unmarshal). The bug hits `*metav1.Time` because it has a custom string converter.